com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v1/ResourceType.java

```diff
@@ -1,32 +1,50 @@
 package com.sap.cloud.lm.sl.cf.core.cf.v1;
 
+import java.util.Collections;
 import java.util.EnumSet;
+import java.util.HashSet;
 import java.util.Set;
 
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+
 public enum ResourceType {
-    MANAGED_SERVICE("managed-service"), USER_PROVIDED_SERVICE("user-provided-service"), EXISTING_SERVICE(
-        "existing-service"), EXISTING_SERVICE_KEY("existing-service-key");
+    MANAGED_SERVICE("managed-service", SupportedParameters.SERVICE, SupportedParameters.SERVICE_PLAN), USER_PROVIDED_SERVICE(
+        "user-provided-service",
+        SupportedParameters.SERVICE_CONFIG), EXISTING_SERVICE("existing-service"), EXISTING_SERVICE_KEY("existing-service-key");
 
-    private String value;
+    private String resourceName;
+    private final Set<String> serviceRequiredParameters = new HashSet<>();
 
-    ResourceType(String value) {
-        this.value = value;
+    private ResourceType(String name, String... requiredParameters) {
+        this.resourceName = name;
+        for (String supportedParameters : requiredParameters) {
+            serviceRequiredParameters.add(supportedParameters);
+        }
     }
 
     @Override
     public String toString() {
-        return value;
+        return resourceName;
     }
 
-    public static ResourceType get(String value) {
-        for (ResourceType v : values()) {
-            if (v.value.equals(value))
-                return v;
+    public static ResourceType get(String resourceType) {
+        for (ResourceType value : ResourceType.values()) {
+            if (value.resourceName.equals(resourceType))
+                return value;
         }
         return null;
     }
 
     public static Set<ResourceType> getServiceTypes() {
         return EnumSet.of(MANAGED_SERVICE, USER_PROVIDED_SERVICE, EXISTING_SERVICE);
     }
+
+    public static Set<String> getRequiredParameters(ResourceType resourceType) {

```

Comment:
Change this to non-static and just return the set of requiredParameters (you have it as a field).